clippy_utils/src/check_proc_macro.rs

```diff
@@ -1,3 +1,4 @@
+#![allow(clippy::impl_trait_in_params)]
```

Comment:
All of these can be reverted. `restriction` lints aren't enabled by default for tests or ci.

Reply:
Fixed.